MinecraftClient/Protocol/Handlers/Protocol18.cs

```diff
@@ -91,85 +98,105 @@ public Protocol18Handler(TcpClient Client, int protocolVersion, IMinecraftComHan
+            else if (protocolVersion >= MC_1_15_Version)
+                entityPalette = new EntityPalette115();
+            else if (protocolVersion >= MC_1_14_Version)
+                entityPalette = new EntityPalette114();
+            else if (protocolVersion >= MC_1_13_Version)
+                entityPalette = new EntityPalette113();
+            else 
+                entityPalette = new EntityPalette112();
 
             // Item palette
-            if (protocolVersion >= MC_1_16_2_Version)
-            {
-                if (protocolVersion > MC_1_19_Version && handler.GetInventoryEnabled())
-                    throw new NotImplementedException(Translations.Get("exception.palette.item"));
-
-                if (protocolVersion == MC_1_19_Version)
-                    itemPalette = new ItemPalette119();
-                else if (protocolVersion >= MC_1_18_1_Version)
-                    itemPalette = new ItemPalette118();
-                else if (protocolVersion >= MC_1_17_Version)
-                    itemPalette = new ItemPalette117();
-                else if (protocolVersion >= MC_1_16_2_Version)
-                    if (protocolVersion >= MC_1_16_2_Version)
-                        itemPalette = new ItemPalette1162();
-                    else itemPalette = new ItemPalette1161();
-            }
-            else itemPalette = new ItemPalette115();
+            if (protocolVersion > MC_1_19_2_Version && handler.GetInventoryEnabled())
+                throw new NotImplementedException(Translations.Get("exception.palette.item"));
+
+            if (protocolVersion >= MC_1_19_Version)
+                itemPalette = new ItemPalette119();
+            else if (protocolVersion >= MC_1_18_1_Version)
+                itemPalette = new ItemPalette118();
+            else if (protocolVersion >= MC_1_17_Version)
+                itemPalette = new ItemPalette117();
+            else if (protocolVersion >= MC_1_16_2_Version)
+                itemPalette = new ItemPalette1162();
+            else if (protocolVersion >= MC_1_16_1_Version)
+                itemPalette = new ItemPalette1161();
+            else 
+                itemPalette = new ItemPalette115();
+
+            // MessageType 
+            // You can find it in https://wiki.vg/Protocol#Player_Chat_Message or /net/minecraft/network/message/MessageType.java
+            if (this.protocolVersion >= MC_1_19_2_Version)
+                ChatParser.ChatId2Type = new()
+                {
+                    { 0,  ChatParser.MessageType.CHAT },
+                    { 1,  ChatParser.MessageType.SAY_COMMAND },
+                    { 2,  ChatParser.MessageType.MSG_COMMAND_INCOMING },
+                    { 3,  ChatParser.MessageType.MSG_COMMAND_OUTGOING },
+                    { 4,  ChatParser.MessageType.TEAM_MSG_COMMAND_INCOMING },
+                    { 5,  ChatParser.MessageType.TEAM_MSG_COMMAND_OUTGOING },
+                    { 6,  ChatParser.MessageType.EMOTE_COMMAND },

```

Comment:
Tested and crashed because chat type 7 (RAW_MSG) isn't included for > 760.
![Screenshot](https://user-images.githubusercontent.com/30041127/188244132-98fafaae-a208-440c-804b-72c8718d199b.png)

Reply:
What form will this message be displayed in the vanilla client, and can I get the server address?
Only these six message types are preconfigured in minecraft's source code. RAW_MSG was added manually for compatibility with 1.19, and it should not be used in 1.19.1+.

Reply:
Tested that chat is presented as a system message when using the tellraw command.
But in [RegistryCodec‘s default value](https://gist.github.com/rj00a/f2970a8ce4d09477ec8f16003b9dce86) you can find the id 7 in "minecraft:chat_type" and its type is "minecraft:tellraw_command".

Added in https://github.com/MCCTeam/Minecraft-Console-Client/pull/2123/commits/2c5b444ffdf9f8c33dc899a5b39aa96a52a24a9b